src/xdotoolify.js

```diff
@@ -240,114 +252,154 @@ _Xdotoolify.prototype.mousemove = function(selector, relpos, twoStep, timeout) {
     selector: selector,
     relpos: relpos.toLowerCase(),
     twoStep: twoStep || false,
-    timeout: timeout || this.defaultTimeout
+    timeout: timeout || this.defaultTimeout,
+    checkAfter: checkAfter
   });
   return this;
 };
-_Xdotoolify.prototype.click = function(mouseButton) {
-  mouseButton = mouseButton || 'left';
+_Xdotoolify.prototype.click = function(mouseButton='left', checkAfter = true) {
   if (!MOUSE_BUTTON_MAPPING[mouseButton.toLowerCase()]) {
     throw new Error('Unknown mouse button ' + mouseButton);
   }
   this._addOperation({
     type: 'click',
     mouseButton: MOUSE_BUTTON_MAPPING[mouseButton.toLowerCase()],
+    checkAfter: checkAfter
   });
   return this;
 };
-_Xdotoolify.prototype.mousedown = function(mouseButton) {
+_Xdotoolify.prototype.mousedown = function(mouseButton, checkAfter = true) {
   mouseButton = mouseButton || 'left';
   if (!MOUSE_BUTTON_MAPPING[mouseButton.toLowerCase()]) {
     throw new Error('Unknown mouse button ' + mouseButton);
   }
   this._addOperation({
     type: 'mousedown',
     mouseButton: MOUSE_BUTTON_MAPPING[mouseButton.toLowerCase()],
+    checkAfter: checkAfter
   });
   return this;
 };
-_Xdotoolify.prototype.mouseup = function(mouseButton) {
+_Xdotoolify.prototype.mouseup = function(mouseButton, checkAfter = true) {
   mouseButton = mouseButton || 'left';
   if (!MOUSE_BUTTON_MAPPING[mouseButton.toLowerCase()]) {
     throw new Error('Unknown mouse button ' + mouseButton);
   }
   this._addOperation({
     type: 'mouseup',
     mouseButton: MOUSE_BUTTON_MAPPING[mouseButton.toLowerCase()],
+    checkAfter: checkAfter
   });
   return this;
 };
-_Xdotoolify.prototype.wheeldown = function() {
-  this.click('wheeldown');
+_Xdotoolify.prototype.wheeldown = function(checkAfter = true) {
+  this.click('wheeldown', checkAfter);
   return this;
 };
-_Xdotoolify.prototype.wheelup = function() {
-  this.click('wheelup');
+_Xdotoolify.prototype.wheelup = function(checkAfter = true) {
+  this.click('wheelup', checkAfter);
   return this;
 };
-_Xdotoolify.prototype.drag = function(selector, mouseButton, timeout) {
-  this.mousedown(mouseButton);
-  this.mousemove(selector, 'center', true, timeout || this.defaultTimeout);
-  this.mouseup(mouseButton);
+_Xdotoolify.prototype.drag = function(
+  selector,
+  mouseButton,
+  timeout,
+  checkAfter = true
+) {
+  this.mousedown(mouseButton, false);
+  this.mousemove(selector, 'center', true, timeout || this.defaultTimeout, false);
```

Comment:
If you need to avoid checkAfter just for this internal api, perhaps we can instead add a second method `_mousemove` that is only to be used for these internal calls. That way the `checkAfter` is not exposed in the public interface in tests.

Reply:
It looks like you need this for click and type as well. So maybe it would be good to prefix `_` in front of all existing methods and then add a corresponding non-`_` version for each that just calls it with `checkAfter=true`

Reply:
I made changes according to both comments. Now they all functions have safe and unsafe versions, with `_` prefixing the unsafe ones.